src/buf_encoder.rs

```diff
@@ -251,6 +251,19 @@ mod tests {
         assert_eq!(encoder.as_str(), "2a");
     }
 
+    #[test]
+    fn put_filler() {
+        let mut encoder = BufEncoder::<8>::new(Case::Lower);
+        assert_eq!(encoder.put_filler(' ', 0), 0);
+        assert_eq!(encoder.as_str(), "");
+        assert_eq!(encoder.put_filler('a', 1), 1);
+        assert_eq!(encoder.as_str(), "a");
+        assert_eq!(encoder.put_filler('é', 2), 2); // Test 2 byte UTF-8
+        assert_eq!(encoder.as_str(), "aéé");
+        assert_eq!(encoder.put_filler('é', 4), 1); // Try to fill more than fits
+        assert_eq!(encoder.as_str(), "aééé");
```

Comment:
Oh, this made me realize our test that `BufEncoder` has even length is likely useless because someone might need to fill in odd number of bytes. :facepalm:

Reply:
I wasn't sure why in the code `buf_encoder.rs:136` it has `let max_capacity = self.buf.remaining_capacity() / filler.len();` when for hex `filler` (char) always has a length 1.  I assumed there must be use case for non hex UTF-8 characters, or at least the code was made to work with them.

Reply:
The `BufEncoder` capacity in characters must be even, not the size in bytes.  I though you were talking about the two byte non hex character in the test at first.